### ISSUE TITLE
[dst] write header at end of every run (Trac #1383)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1375]
A rare edge case came up where the last subrun had less frames than the prescale, so was missing a dst header. The solution is to just write a header at the end of every run.

Implementation hints:
- probably need a buffer of 1 frame, so you can modify the last frame and push it on receiving a finish

Note: an alternative would be to change the MultiWriter so you could get a signal whenever switching to a new file, but this is unlikely to happen.

Migrated from https://code.icecube.wisc.edu/ticket/1383
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:15:18", 
    "description": "A rare edge case came up where the last subrun had less frames than the prescale, so was missing a dst header. The solution is to just write a header at the end of every run.\n\nImplementation hints:\n- probably need a buffer of 1 frame, so you can modify the last frame and push it on receiving a finish\n\n\nNote: an alternative would be to change the MultiWriter so you could get a signal whenever switching to a new file, but this is unlikely to happen.", 
    "reporter": "david.schultz", 
    "cc": "blaufuss", 
    "resolution": "wontfix", 
    "_ts": "1550067318169976", 
    "component": "combo reconstruction", 
    "summary": "[dst] write header at end of every run", 
    "priority": "major", 
    "keywords": "", 
    "time": "2015-10-05T16:39:39", 
    "milestone": "", 
    "owner": "tschmidt", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20151005T17:21:48: 
* david.schultz commented:

> Apparently there is an issue with this implementation and PnF.  Plan on discussing at fall collaboration meeting.


* david.schultz changed cc from "" to "blaufuss"

**Comment by icecube-trac** [comment]
Trac update at 20160420T18:52:16: olivas changed milestone from "March 2016" to "April 2016"

**Comment by icecube-trac** [comment]
Trac update at 20170405T16:12:11: juancarlos changed milestone from "May 2016" to "July 2016"

**Comment by icecube-trac** [comment]
Trac update at 20170407T18:28:26: juancarlos changed milestone from "July 2016" to "October 2017"

**Comment by icecube-trac** [comment]
Trac update at 20180607T19:28:31: olivas changed milestone from "October 2017" to "Summer Solstice 2018"

**Comment by icecube-trac** [comment]
Trac update at 20180725T13:22:57: olivas changed milestone from "Summer Solstice 2018" to "Autumnal Equinox 2018"

**Comment by icecube-trac** [comment]
Trac update at 20180730T15:38:13: 
* tschmidt changed status from "new" to "assigned"
* tschmidt changed owner from "juancarlos" to "tschmidt"

**Comment by icecube-trac** [comment]
Trac update at 20180730T15:47:28: tschmidt commented:

> the problem is that dst data is created within the filter clients in pnf.
> so, there is O(100) dst modules running in parallel distributed over the network.
> even if one "sends a signal" from the writers, how would a client know: i am the one to insert the header. also, i don't like this "noise" on the network - there is enough load to it already. beside: pnf has many buffers and these can be quiet big. so, even if this client would know that it is the one, the relevant frames/events may have passed already anyway.
> i am also not sure, if i like a "one frame buffer" thing within the server...
> i wonder, if we could just increase the dst header prescale - a question of load and bandwidth, we may play with this setting.
> i also wonder: this data is a stream, so why can't offline keep the latest header from file (n-1) to re-use it in file n. the splitting into files is an arbitrary decision anyway and we have several analysis that depend on the time dependency/sorting anyway, right? Juan-Carlos told me he will check this option.
> anway: both solutions are no real fixes of this ticket. i will close it "wontfix".

**Comment by icecube-trac** [comment]
Trac update at 20180730T15:47:58: 
* tschmidt changed resolution from "" to "wontfix"
* tschmidt changed status from "assigned" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:15:18: 
* olivas commented:

> Milestone Autumnal Equinox 2018 deleted


* olivas changed milestone from "Autumnal Equinox 2018" to ""
